recbole/evaluator/collector.py

```diff
@@ -84,14 +84,14 @@ def data_collect(self, train_data):
         """
         if self.register.need("data.num_items"):
             item_id = self.config["ITEM_ID_FIELD"]
-            self.data_struct.set("data.num_items", train_data.dataset.num(item_id))
+            self.data_struct.set("data.num_items", train_data._dataset.num(item_id))
         if self.register.need("data.num_users"):
             user_id = self.config["USER_ID_FIELD"]
-            self.data_struct.set("data.num_users", train_data.dataset.num(user_id))
+            self.data_struct.set("data.num_users", train_data._dataset.num(user_id))
         if self.register.need("data.count_items"):
-            self.data_struct.set("data.count_items", train_data.dataset.item_counter)
+            self.data_struct.set("data.count_items", train_data._dataset.item_counter)
         if self.register.need("data.count_users"):
-            self.data_struct.set("data.count_users", train_data.dataset.user_counter)
+            self.data_struct.set("data.count_items", train_data._dataset.user_counter)
```

Comment:
Maybe line 94 is `self.data_struct.set("data.count_users", train_data._dataset.user_counter)`, since `need("data.count_users")`.

Reply:
It is probably correct, but in the version I downloaded by using `pip install recbole==1.1.1 dgl`
line 94 is
`self.data_struct.set("data.count_items", train_data.dataset.user_counter)`

I also checked the source code on my personal laptop and I can confirm that.

Reply:
@giuspillo Yes, it is indeed an existing Typo, but we also hope that you can change it together so that we don't have to open a new PR to fix this issue. Thanks very much for your support and enhancement to RecBole!

Reply:
Line 94 fixed and committed :)